drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2163 @@
+ *    VBUS/VCONN overcurrent protection, VBUS overvoltage protection, etc.,
+ *    can implement all functions defined in TCPC, PPC, and VBUS. For this,
+ *    TCPC is implemented in UTCPD majorly; PPC and VBUS rely on TCPC for
+ *    their implementation.
+ * 2. For VBUS/VCONN voltage measurement, UTCPD is updated periodically
+ *    by Timer-trigger EADC. To implement this interconnection, TCPC node_id
+ *    will cover UTCPD, EADC, and Timer H/W characteristics of registers,
+ *    interrupts, resets, and clocks.
+ *    NOTE: EADC and Timer interrupts needn't enable for Timer-triggered EADC.
+ *          In BSP sample, they are enabled just for development/debug purpose.
+ * 3. About VCONN per PCB
+ *    (1) Support only VCONN source, no VCONN sink (like Plug Cable)
+ *    (2) Separate pins for VCONN enable on CC1/CC2 (VCNEN1/VCNEN2)
+ *    (3) Single pin for VCONN discharge (DISCHG)
+ * 4. VBUS discharge precedence
+ *    (1) GPIO
+ *    (2) UTCPD
+ * 5. VCONN discharge precedence
+ *    (1) DPM-supplied callback
+ *    (2) GPIO
+ *    (3) UTCPD
+ */
+
+/* Invalid or missing value */
+#define NUMAKER_INVALID_VALUE ((uint32_t)-1)
+
+/* UTCPD VBUS threshold default in mV
+ *
+ * These are default values of UTCPD VBUS threshold registers. They need
+ * to be reconfigured by taking the following factors into consideration:
+ * 1. Analog Vref
+ * 2. UTCPD VBVOL.VBSCALE
+ */
+#define NUMAKER_UTCPD_VBUS_THRESHOLD_OVERVOLTAGE          25000
+#define NUMAKER_UTCPD_VBUS_THRESHOLD_VSAFE5V              5000
+#define NUMAKER_UTCPD_VBUS_THRESHOLD_VSAFE0V              0
+#define NUMAKER_UTCPD_VBUS_THRESHOLD_STOP_FORCE_DISCHARGE 800
+#define NUMAKER_UTCPD_VBUS_THRESHOLD_SINK_DISCONNECT      3500
+
+/* This goto replacement is to silence checkpatch complaint:
+ * WARNING:MACRO_WITH_FLOW_CONTROL: Macros with flow control statements should be avoided
+ *
+ * Use of goto for single exit point (cleanup) is decent. Here, its use
+ * can make code e.g. UTCPD register write with error condition more concise.
+ */
+#define NUMAKER_GOTO UTIL_CAT(go, to)
+
+/* Helper which makes the boilerplate code for 'rc' expression check */
+#define NUMAKER_RC_CHECK(rc_expr)                                                                  \
+	rc = (rc_expr);                                                                            \
+	if (rc < 0) {                                                                              \
+		NUMAKER_GOTO cleanup;                                                              \
+	}
+#define NUMAKER_RC_CHECK_WITH_MSG(rc_expr, msg_on_err)                                             \
+	rc = (rc_expr);                                                                            \
+	if (rc < 0) {                                                                              \
+		LOG_ERR(msg_on_err);                                                               \
+		NUMAKER_GOTO cleanup;                                                              \
+	}
```

Comment:
this is very non zephyresque and seriously hacky, really just check manually after every function, I know it's many location and the lines add up but this is how it's done in the rest of the code base

Reply:
As suggested, move `goto` outside of macro and check it manually

Reply:
thanks